src/routes/dashboard/containers/DashboardContainer.js

```diff
@@ -70,14 +70,18 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
     return { status: 'routeNotInitialized' }
   }
 
-  if (state.cities.status === 'loading' || switchingLanguage || route.status === 'loading' || !languages) {
+  if (state.cities.status === 'loading' || switchingLanguage || route.status === 'loading' ||
+    languages.status === 'loading') {
     return { status: 'loading' }
   }
 
   if (route.status === 'languageNotAvailable') {
+    if (languages.status !== 'ready') {
+      throw Error('languageNotAvailable status impossible if languages not ready')
```

Comment:
We should never throw here! Rather return a general error or something different.